equinix/resource_network_bgp_test.go

```diff
@@ -100,6 +101,20 @@ func (r *mockedBGPUpdateRequest) Execute() error {
 	return nil
 }
 
+var _ packngo.BGPConfigService = &mockBgpConfigService{}
```

Comment:
If we do stick with mocking the SDK (I still think we shouldn't, but it sounds like maybe we will), this mock should be in a different file.  The `resource_network_bgp_test.go` file is a test file for a Network Edge resource, not a Metal resource.